internal/flow/release.go

```diff
@@ -104,117 +29,14 @@ func releaseConfigurationExists(resourcePath string) error {
-		logger.Infof("flow: ReleaseBranch: copy resources from %s to %s", artifactPath, destinationPath)
-
-		err = s.cleanCopy(ctx, artifactPath, destinationPath)
-		if err != nil {
-			return true, errors.WithMessagef(err, "copy resources from '%s' to '%s'", artifactPath, destinationPath)
-		}
-
-		// copy artifact spec
-		// repo/{env}/releases/{ns}/{service}/{artifactFileName}
-		artifactDestinationPath := path.Join(releasePath(sourceConfigRepoPath, service, environment, namespace), s.ArtifactFileName)
-		logger.Infof("flow: ReleaseBranch: copy artifact from %s to %s", artifactSpecPath, artifactDestinationPath)
-		err = copy.CopyFile(ctx, artifactSpecPath, artifactDestinationPath)
-		if err != nil {
-			return true, errors.WithMessage(err, fmt.Sprintf("copy artifact spec from '%s' to '%s'", artifactSpecPath, artifactDestinationPath))
-		}
-
-		artifactSpec, err := s.Storage.LatestArtifactSpecification(ctx, service, branch)
-		if err != nil {
-			return true, errors.WithMessage(err, "get artifact spec")
-		}
-
-		authorName := artifactSpec.Application.AuthorName
-		authorEmail := artifactSpec.Application.AuthorEmail
-		artifactID := artifactSpec.ID
-		releaseMessage := git.ReleaseCommitMessage(environment, service, artifactID, authorEmail)
-		err = s.Git.Commit(ctx, sourceConfigRepoPath, releasePath(".", service, environment, namespace), authorName, authorEmail, actor.Name, actor.Email, releaseMessage)
-		if err != nil {
-			if errors.Cause(err) == git.ErrNothingToCommit {
-				logger.Infof("Environment is up to date: dropping event: %v", err)
-				// TODO: notify actor that there was nothing to commit
-				return true, nil
-			}
-			// we can see races here where other changes are committed to the master repo
-			// after we cloned. Because of this we retry on any error.
-			return false, errors.WithMessage(err, fmt.Sprintf("commit changes from path '%s'", destinationPath))
-		}
-		s.notifyRelease(ctx, NotifyReleaseOptions{
-			Service:     service,
-			Environment: environment,
-			Namespace:   namespace,
-			Spec:        artifactSpec,
-			Releaser:    actor.Name,
-		})
-		logger.Infof("flow: ReleaseBranch: release committed: %s, Author: %s <%s>, Committer: %s <%s>", releaseMessage, authorName, authorEmail, actor.Name, actor.Email)
-		return true, nil
-	})
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
 type ReleaseArtifactIDEvent struct {
-	Service     string `json:"service,omitempty"`
-	Environment string `json:"environment,omitempty"`
-	Namespace   string `json:"namespace,omitempty"`
-	ArtifactID  string `json:"artifactID,omitempty"`
-	Branch      string `json:"branch,omitempty"`
-	Actor       Actor  `json:"actor,omitempty"`
+	Service     string        `json:"service,omitempty"`
+	Environment string        `json:"environment,omitempty"`
+	Namespace   string        `json:"namespace,omitempty"`
+	ArtifactID  string        `json:"artifactID,omitempty"`
+	Branch      string        `json:"branch,omitempty"`
```

Comment:
Is the `Branch` field still used now or is it just kept around for easy printing in the async flow?

Reply:
Thought we could remove it, but it is used by `storage.ArtifactPaths` 😔
I think we can remove it later on when we pension git artifact storage.